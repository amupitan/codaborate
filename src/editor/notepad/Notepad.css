.notepad-editor{
    outline: none;
    background-color: #000000;
    color: #ffffff;
    width: 100%;
    height: 100%;
}