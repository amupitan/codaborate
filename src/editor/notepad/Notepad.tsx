import * as React from 'react';
import './Notepad.css';

export interface ITheme {
    backgroundColor?: string;
    color?: string;
}

export interface IProps extends ITheme {
    content?: string;
}

const Notepad: React.SFC<IProps> = (props: IProps) => {
    const { content, ...styles } = props;
    return (
        <div className='notepad-editor' style={styles} contentEditable={true} suppressContentEditableWarning={true} >
            {content}
        </div>
    );
};

const defaultTheme: ITheme = {
    backgroundColor: '#000000',
    color: '#ffffff',
};

Notepad.defaultProps = {
    ...defaultTheme,
    content: 'Hello World! ðŸ˜€',
};

export default Notepad;


