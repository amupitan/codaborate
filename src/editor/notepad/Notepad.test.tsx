import { mount, shallow } from 'enzyme';
import * as React from 'react';

import '../../setupTests';
import Notepad from './Notepad';

describe('Notepad', () => {
  it('renders without crashing', () => {
    const component = mount(<Notepad />);
    component.unmount();
  });

  it('should render correctly with no props', () => {
    const component = shallow(<Notepad />);

    expect(component).toMatchSnapshot();
  });

  it('should render content', () => {
    const content = 'this is some content ðŸ˜Ž';
    const component = shallow(<Notepad content={content} />);

    expect(component).toMatchSnapshot();
  });
});