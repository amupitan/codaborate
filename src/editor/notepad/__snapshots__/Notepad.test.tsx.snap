// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notepad should render content 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Notepad
    backgroundColor="#000000"
    color="#ffffff"
    content="this is some content ðŸ˜Ž"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "this is some content ðŸ˜Ž",
      "className": "notepad-editor",
      "contentEditable": true,
      "style": Object {
        "backgroundColor": "#000000",
        "color": "#ffffff",
      },
      "suppressContentEditableWarning": true,
    },
    "ref": null,
    "rendered": "this is some content ðŸ˜Ž",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "this is some content ðŸ˜Ž",
        "className": "notepad-editor",
        "contentEditable": true,
        "style": Object {
          "backgroundColor": "#000000",
          "color": "#ffffff",
        },
        "suppressContentEditableWarning": true,
      },
      "ref": null,
      "rendered": "this is some content ðŸ˜Ž",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Notepad should render correctly with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Notepad
    backgroundColor="#000000"
    color="#ffffff"
    content="Hello World! ðŸ˜€"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Hello World! ðŸ˜€",
      "className": "notepad-editor",
      "contentEditable": true,
      "style": Object {
        "backgroundColor": "#000000",
        "color": "#ffffff",
      },
      "suppressContentEditableWarning": true,
    },
    "ref": null,
    "rendered": "Hello World! ðŸ˜€",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hello World! ðŸ˜€",
        "className": "notepad-editor",
        "contentEditable": true,
        "style": Object {
          "backgroundColor": "#000000",
          "color": "#ffffff",
        },
        "suppressContentEditableWarning": true,
      },
      "ref": null,
      "rendered": "Hello World! ðŸ˜€",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
